tools/cli/domain_utils.go

```diff
@@ -51,173 +51,197 @@ const (
+		},
+		&cli.StringFlag{
+			// TODO use StringFlag instead of buggy StringSliceFlag
+			Name:    FlagClusters,
+			Aliases: []string{"cl"},
+			Usage:   "Clusters",
+		},
+		&cli.StringFlag{
+			Name:    FlagIsGlobalDomain,
+			Aliases: []string{"gd"},
+			Usage:   "Flag to indicate whether domain is a global domain. Default to true. Local domain is now legacy.",
+			Value:   "true",
+		},
+		&cli.GenericFlag{
+			Name:    FlagDomainData,
+			Aliases: []string{"dmd"},
+			Usage:   "Domain data of key value pairs (must be in key1=value1,key2=value2,...,keyN=valueN format, e.g. cluster=dca or cluster=dca,instance=cadence)",
+			Value:   &flag.StringMap{},
+		},
+		&cli.StringFlag{
+			Name:    FlagSecurityToken,
+			Aliases: []string{"st"},
+			Usage:   "Optional token for security check",
+		},
+		&cli.StringFlag{
+			Name:    FlagHistoryArchivalStatus,
+			Aliases: []string{"has"},
+			Usage:   "Flag to set history archival status, valid values are \"disabled\" and \"enabled\"",
+		},
+		&cli.StringFlag{
+			Name:    FlagHistoryArchivalURI,
+			Aliases: []string{"huri"},
+			Usage:   "Optionally specify history archival URI (cannot be changed after first time archival is enabled)",
+		},
+		&cli.StringFlag{
+			Name:    FlagVisibilityArchivalStatus,
+			Aliases: []string{"vas"},
+			Usage:   "Flag to set visibility archival status, valid values are \"disabled\" and \"enabled\"",
+		},
+		&cli.StringFlag{
+			Name:    FlagVisibilityArchivalURI,
+			Aliases: []string{"vuri"},
+			Usage:   "Optionally specify visibility archival URI (cannot be changed after first time archival is enabled)",
 		},
 	}
 
 	updateDomainFlags = []cli.Flag{
-		cli.StringFlag{
-			Name:  FlagDescriptionWithAlias,
-			Usage: "Domain description",
-		},
-		cli.StringFlag{
-			Name:  FlagOwnerEmailWithAlias,
-			Usage: "Owner email",
-		},
-		cli.StringFlag{
-			Name:  FlagRetentionDaysWithAlias,
-			Usage: "Workflow execution retention in days",
-		},
-		cli.StringFlag{
-			Name:  FlagActiveClusterNameWithAlias,
-			Usage: "Active cluster name",
-		},
-		cli.StringFlag{
+		&cli.StringFlag{
+			Name:    FlagDescription,
+			Aliases: []string{"desc"},
+			Usage:   "Domain description",
+		},
+		&cli.StringFlag{
+			Name:    FlagOwnerEmail,
+			Aliases: []string{"oe"},
+			Usage:   "Owner email",
+		},
+		&cli.StringFlag{
+			Name:    FlagRetentionDays,
+			Aliases: []string{"rd"},
+			Usage:   "Workflow execution retention in days",
+		},
+		&cli.StringFlag{
+			Name:    FlagActiveClusterName,
+			Aliases: []string{"ac"},
+			Usage:   "Active cluster name",
+		},
+		&cli.StringFlag{
 			// use StringFlag instead of buggy StringSliceFlag
-			// TODO when https://github.com/urfave/cli/pull/392 & v2 is released
-			//  consider update urfave/cli
-			Name:  FlagClustersWithAlias,
-			Usage: "Clusters",
-		},
-		cli.GenericFlag{
-			Name:  FlagDomainDataWithAlias,
+			// TODO when https://github.com/urfave/cli/v2/pull/392 & v2 is released
+			//  consider update urfave/cli/v2
```

Comment:
probably update to match the other (and perhaps update both to not-imply StringSliceFlag is _currently_ buggy, as I think it is not?)

we can migrate later, just seems worth updating the comments.

Reply:
update the comment. We need to use StringSlice. Current implementation is error prone